orangecontrib/prototypes/widgets/owsplit.py

```diff
@@ -0,0 +1,135 @@
+import Orange.data
+import numpy as np
+from Orange.data import Table
+from Orange.data.util import SharedComputeValue
+from Orange.widgets import gui
+from Orange.widgets.settings import ContextSetting, DomainContextHandler
+from Orange.widgets.widget import OWWidget, Msg, Output, Input
+from AnyQt.QtCore import Qt
+
+from Orange.data import DiscreteVariable, StringVariable
+from Orange.widgets.utils.itemmodels import DomainModel
+from Orange.widgets.utils.widgetpreview import WidgetPreview
+
+foo = ["foo,bar", "foo,baz", "foo", "bar,baz", "baz,foo", "baz", "bar"]
+
+
+def values_from_string(var_list, splitter):
+    values = [s.split(splitter) for s in var_list]
+    new_values = sorted(set(val for vals in values for val in vals))
+    mapper = {v: i for i, v in enumerate(new_values)}
+
+    X = np.zeros((len(var_list), len(new_values)))
+    for i, vals in enumerate(values):
+        X[i, list(map(mapper.get, vals))] = 1
+    return X, new_values
+
+
+def values_to_string(var_list, values):
+    # turn discrete to string variable
+    return [values[int(i)] if not np.isnan(i) else "nan" for i in var_list]
+
+
+class TransformDomain:
```

Comment:
As far as I see, this class is not used. Neither necessary?

Reply:
Is this not necessary for compute_value?